### PR TITLE
rust: add Tonic version of generated Rust code

### PR DESCRIPTION
**Pull request by hawkw**
Tonic (https://crates.io/crates/tonic) is, essentially, the Tokio 0.2
world's version of `tower-grpc`. Although the two libraries are similar
and share a common geneology, the code generated by `tonic-build` is not
compatible with the code generated by `tower-grpc-build`. In order to
implement the mock destination controller for linkerd/linkerd2#4158
using the Tokio 0.2 stack, we need a Tonic version of the generated Rust
bindings.

This branch adds a new `linkerd2-proxy-api-tonic` crate that contains
Tonic bindings generated from the proxy-api protos. Rather than putting
this code in the existing `linkerd2-proxy-api` crate under a feature
flag, I opted to create a separate crate. This is because `tonic-build`
and `tower-grpc-build` will generate code that depends on different
versions of libraries like `tower`, `http`, and `prost`. Although it
would be possible to rename those dependencies so we can depend on both
their Tokio 0.1 and Tokio 0.2 versions, it seems significantly more
difficult to convince the code generators to generate code that
references those dependencies by names like `http_02` and so on.
Therefore, I created a separate crate for the Tonic version.

This may also be useful when we start updating the proxy itself to use
the Tokio 0.2 stack.

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

**Comment by olix0r** [comment]
@hawkw there's no urgency on this, but what do you think about taking a branch-based approach for tokio-0.2 (as we did in the proxy repo)?

**Comment by hawkw** [comment]
@olix0r sure, I'd be fine to change to that approach here.

File: Cargo.toml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [workspace]
 members = [
-    "rs"
+    "rs",
+    "rs-tonic",
```
Comment (olix0r) [inline Cargo.toml:4]: fwiw, I'm in favor of just replacing the `rs` crate

Comment (hawkw) [reply Cargo.toml:4]: yup, will do

File: rs/build.rs [hunk 2]
```diff
@@ -1,27 +1,28 @@
-extern crate tower_grpc_build;
-
-fn main() {
-    build_control();
+use std::error::Error;
+fn main() -> Result<(), Box<dyn Error>> {
```
Comment (olix0r) [inline rs/build.rs:2]:
```suggestion

fn main() -> Result<(), Box<dyn Error>> {
```

**Pull request merged**